src/decrypters/widevine/WVDecrypter.cpp

```diff
@@ -26,6 +34,49 @@ CWVDecrypter::~CWVDecrypter()
   m_WVCdmAdapter = nullptr;
 }
 
+bool CWVDecrypter::Initialize()
+{
+#if defined(__linux__)
+#if defined(__aarch64__) || defined(__arm64__)
+  // On linux arm64, libwidevinecdm.so depends on two dynamic symbols:
+  //   __aarch64_ldadd4_acq_rel
+  //   __aarch64_swp4_acq_rel
+  // These are defined from a separate library cdm_aarch64_loader,
+  // but to make them available in the main binary's PLT, we need RTLD_GLOBAL.
+  // Kodi kodi::tools::CDllHelper LoadDll() cannot be used because use RTLD_LOCAL,
+  // and we need the RTLD_GLOBAL flag.
+  std::string binaryPath;
+  std::vector<kodi::vfs::CDirEntry> items;
+  if (kodi::vfs::GetDirectory(FILESYS::GetAddonPath(), "", items))
+  {
+    for (auto item : items)
+    {
+      if (!STRING::Contains(item.Label(), "cdm_aarch64_loader"))
+        continue;
+
+      binaryPath = item.Path();
+      break;
+    }
+  }
+  if (binaryPath.empty())
+  {
+    LOG::Log(LOGERROR, "Cannot find the cdm_aarch64_loader file");
+    return false;
+  }
+
+  void* hdl = dlopen(binaryPath.c_str(), RTLD_GLOBAL | RTLD_LAZY);
```

Comment:
Are you missing `RTLD_NOLOAD`?

Reply:
just tested and now this error is shown:
error <general>: AddOnLog: inputstream.adaptive: Initialize: Failed to load CDM aarch64 loader from path "/storage/.kodi/addons/inputstream.adaptive/libcdm_aarch64_loader.so", error: (null)

The "libcdm_aarch64_loader.so" of course exists in the desired folder.

Reply:
@DDuck007 please dont comment on threads of other users when not related to the highlighted code

i have not added RTLD_NOLOAD because it is used when a library has been already previously loaded,
on old code the library was previously loaded by LoadDll
https://github.com/xbmc/inputstream.adaptive/pull/1166/files#diff-a1accd572cc063318954b05640f5f8393d536b6c95bcb6e2535d873aeb3e0cccR141
but here is never opened

Reply:
What do you mean? Isn't inputstream.adaptive itself already dlopened from Kodi?

Reply:
flag added, if someone can test it

but from linux docs seem explain different
from code in this way this binary is never loaded at runtime, where old code was dloaded twice times one time regular dload and second time with RTLD_NOLOAD/GLOBAL,
or maybe you means that executing it with RTLD_NOLOAD will influence also the symbols also of "parent" inputstream library?